ledger/sandbox-common/src/test/lib/scala/platform/sandbox/services/reset/ResetServiceITBase.scala

```diff
@@ -76,30 +82,43 @@ abstract class ResetServiceITBase
   override protected def config: SandboxConfig =
     super.config.copy(ledgerIdMode = LedgerIdMode.Dynamic)
 
-  protected val eventually: RetryStrategy = RetryStrategy.exponentialBackoff(10, scaled(10.millis))
+  protected val eventually: RetryStrategy = RetryStrategy.constant(50, 100.milliseconds)
+
+  protected implicit val ec: ExecutionContext = ExecutionContext.global
 
   override protected def darFile: File =
     new File(rlocation("ledger/test-common/model-tests.dar"))
 
   protected def timeIsStatic: Boolean =
     config.timeProviderType.getOrElse(SandboxConfig.DefaultTimeProviderType) == TimeProviderType.Static
 
-  protected def fetchLedgerId(): Future[String] =
+  protected def waitForLedgerToStart(): Future[LedgerId] =
+    for {
+      ledgerId <- eventually { (_, _) =>
+        fetchLedgerId()
+      }
+      // Completions won't work until a ledger configuration is in place, so we wait for one.
+      _ <- new StreamConsumer[GetLedgerConfigurationResponse](responseObserver =>
+        LedgerConfigurationServiceGrpc
+          .stub(channel)
+          .getLedgerConfiguration(GetLedgerConfigurationRequest(ledgerId.unwrap), responseObserver))
+        .firstWithin(Span.convertSpanToDuration(scaled(1.second)))
```

Comment:
If I understand correctly, `firstWithin` will complete after a maximum of 1 second, even if the ledger configuration is not ready. Maybe you should check whether the result of `firstWithin` is not an empty vector, and return a failed promise otherwise?

Reply:
Good call. I'll fix that.

Reply:
Done.